docs/DeveloperGuide.md

```diff
@@ -154,6 +154,123 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+
+### Find Doctor / Patient
+
+**Introduction**
+
+This section describes the feature that allows users to find doctors/patient in the MediConnect database by name.
+
+#### Implementation
+The finding of a doctor/patient in MediConnect is facilitated by LogicManager, which extends Logic. It holds a mediConnectParser that parses the user input, and a Model where the command is executed. Key methods involved include:
+
+LogicManager#execute() — Executes the given user String input and returns a CommandResult
+FindDoctorCommandParser/FindPatientCommandParser#parse() — Parses the user input to create a FindDoctorCommand/FindPatientCommand
+FindDoctorCommand/FindPatientCommand#execute() — Filters the list of doctors/patient based on the given predicate
+These operations are exposed in the Ui interface as Ui#executeCommand().
+
+Given below is an example usage scenario and how the ListDoctorCommand/ListPatientCommand mechanism behaves at each step.
+
+Step 1: The user inputs find_d/find_n John to search for doctors/patients named "John" in MediConnect.
+* The find_d/find_n command triggers mediConnectParser#parseCommand, which identifies the command word and calls FindDoctorCommandParser/FindPatientCommandParser#parse to handle the arguments.
+
+Step 2: The FindDoctorCommandParser/FindPatientCommandParser#parse method splits the argument "John" into a list of keywords. It then creates a NameContainsKeywordsDoctorPredicate/NameContainsKeywordsPatientPredicate object, using the list of keywords.
+
+Step 3: A new FindDoctorCommand/FindPatientCommand instance is created using the NameContainsKeywordsDoctorPredicate/NameContainsKeywordsPatientPredicate object.
+
+Step 4: The created FindDoctorCommand/FindPatientCommand instance is returned to LogicManager, and its execute method is called.
+FindDoctorCommand/FindPatientCommand#execute filters the list of doctors/patients in Model using the NameContainsKeywordsDoctorPredicate/NameContainsKeywordsPatientPredicate.
+
+Step 5: The filtered list is displayed to the user through the UI.
+
+The following sequence diagram shows how the find doctor operation would work: SortSequenceDiagram
+    ![FindDoctorSequence](images/FindDoctorSequence.png)
+
+The following sequence diagram shows how the find patient operation would work: SortSequenceDiagram
+    ![FindPatientSequence](images/FindPatientSequence.png)
+
+The following activity diagram summarizes what happens when a user wants to find a new doctor/patient: FindYZCommandActivityDiagram
```

Comment:
Can remove the Sort SequenceDiagram and FindYZCommandActivityDiagram after the :